dialect/clickhouse.go

```diff
@@ -0,0 +1,76 @@
+package dialect
+
+import (
+	"fmt"
+	"strings"
+	"time"
+)
+
+// clickhouse dialect is based on mysql dialect, but using
+// [...] to represent array instead of (...).
+type clickhouse mysql
+
+func (d clickhouse) QuoteIdent(s string) string {
```

Comment:
Can you embed mysql to avoid duplicating these functions?

Reply:
oops, commit this by mistake. I had defined this as `type clickhouse struct{}` so as to separate it with other dialects clearly.

Reply:
Are these implementations mostly same as mysql? If so we should only implement the functions with different implementations for CH

Reply:
yeah, only differ on EncodeArrayBeing/End(). The existing dialects actually share common functionalities in this way. 
```
func (d postgreSQL) EncodeTime(t time.Time) string {
	return MySQL.EncodeTime(t)
} 
```

Reply:
kk